### PR TITLE
Add key to request the broker data

### PR DESCRIPTION
**Pull request by p3dr0rv**
Add a key to request broker data

We are setting this key in ContentProviderStrategy so all MSAL, MWPJ and Broker API request, can update the broker information.

related PR https://github.com/AzureAD/ad-accounts-for-android/pull/2507

### CODE REVIEW

**Comment by p3dr0rv** [comment]
@rpdome, I wonder if we should also include the legacy WPJ? or just wait for multiple-wpj to be adopted, anyway this is just an optimization for edge cases

File: common/src/main/java/com/microsoft/identity/common/internal/controllers/BrokerOperationExecutor.java [hunk 1]
```diff
@@ -232,6 +234,9 @@ private <T> T performStrategy(@NonNull final IIpcStrategy strategy,
             span.setAttribute(AttributeName.ipc_strategy.name(), strategy.getType().name());
             operation.performPrerequisites(strategy);
             final BrokerOperationBundle brokerOperationBundle = operation.getBundle();
+            if (brokerOperationBundle.getBundle() != null) {
```
Comment (rpdome) [inline common/src/main/java/com/microsoft/identity/common/internal/controllers/BrokerOperationExecutor.java]: if bundle is null, create an empty bundle with the flag?

**Review by rpdome — APPROVED** [review]

**Comment by rpdome** [comment]
> @rpdome, I wonder if we should also include the legacy WPJ? or just wait for multiple-wpj to be adopted, anyway this is just an optimization for edge cases

no need to do for legacy wpj - as it doesn't use the new selection mechanism to begin with.

File: common/src/main/java/com/microsoft/identity/common/internal/controllers/BrokerOperationExecutor.java (hunk 1 above)
Comment (p3dr0rv) [reply common/src/main/java/com/microsoft/identity/common/internal/controllers/BrokerOperationExecutor.java]: bundle is final so I can move it inside brokerOperationBundle

File: common/src/main/java/com/microsoft/identity/common/internal/cache/ActiveBrokerCacheUpdater.kt [hunk 2]
```diff
@@ -71,6 +71,12 @@ class ActiveBrokerCacheUpdater(
          */
         const val BROKER_DISCOVERY_DISABLED_KEY = "broker.discovery.disabled"
 
+        /**
+         * If this key is present in the request bundle,
+         * The broker will return the active broker information in the result bundle.
+         * Note: This is only valid for requests that are processed by the BrokerOperationRequestDispatcher.
+         */
+        const val REQUEST_ACTIVE_BROKER_DATA_KEY = "com.microsoft.identity.request.broker.data"
```
Comment (shahzaibj) [inline common/src/main/java/com/microsoft/identity/common/internal/cache/ActiveBrokerCacheUpdater.kt]: For all such keys, I think renaming them such that name starts with the word "KEY" might be better for readability. For example: `KEY_REQUEST_ACTIVE_BROKER_DATA`

File: common/src/main/java/com/microsoft/identity/common/internal/cache/ActiveBrokerCacheUpdater.kt [hunk 3]
```diff
@@ -125,6 +131,7 @@ class ActiveBrokerCacheUpdater(
         val signingCertThumbprint = bundle.getString(ACTIVE_BROKER_SIGNING_CERTIFICATE_THUMBPRINT_KEY)
 
         if (packageName.isNullOrEmpty() || signingCertThumbprint.isNullOrEmpty()) {
+            Logger.info(methodTag, "A response was received without active broker information.")
```
Comment (shahzaibj) [inline common/src/main/java/com/microsoft/identity/common/internal/cache/ActiveBrokerCacheUpdater.kt:134]: Should we also log if it was requested to begin with?

**Review by shahzaibj — APPROVED** [review]

File: common/src/main/java/com/microsoft/identity/common/internal/cache/ActiveBrokerCacheUpdater.kt (hunk 3 above)
Comment (p3dr0rv) [reply common/src/main/java/com/microsoft/identity/common/internal/cache/ActiveBrokerCacheUpdater.kt:134]: done

**Pull request merged**